core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectLifecycle.scala

```diff
@@ -111,6 +111,54 @@ trait BillingProjectLifecycle extends LazyLogging {
       })
     }
   }
+
+  def deleteBillingProfileAndUnregisterBillingProject(projectName: RawlsBillingProjectName,
+                                                      billingProfileExpected: Boolean,
+                                                      ctx: RawlsRequestContext
+  )(implicit
+    executionContext: ExecutionContext
+  ): Future[Unit] = for {
+    billingProfileId <- billingRepository.getBillingProfileId(projectName)
+    _ <- (billingProfileId, billingProfileExpected) match {
+      case (Some(id), _) => cleanupBillingProfile(UUID.fromString(id), projectName, ctx)
+      case (None, true) =>
+        logger.warn(
+          s"Deleting billing project $projectName that was expected to have a billing profile, but no associated billing profile record to delete"
+        )
+        Future.successful()
+      case (None, false) =>
+        logger.info(
+          s"Deleting billing project $projectName, but no associated billing profile record to delete (could be a legacy project)"
```

Comment:
Once we have backfilled billing profiles for existing GCP billing projects, we can get rid of this case.